hazelcast/src/main/java/com/hazelcast/map/impl/querycache/subscriber/DefaultQueryCache.java

```diff
@@ -334,7 +334,7 @@ public V get(Object key) {
 
         Set<K> resultingSet = new HashSet<K>();
 
-        Set<QueryableEntry> query = indexes.query(predicate);
+        Set<QueryableEntry> query = indexes.query(predicate, -1);
```

Comment:
Is it OK to pass -1 here? It seem that we may cache inconsistent results here if the check for partition count is skipped, don't we?

Reply:
Seems like it's impossible to fix that without introducing changes to query cache itself and that's out of scope of this task. Probably we should just open a new issue. @ahmetmircik wdyt?

Reply:
I've created an issue https://github.com/hazelcast/hazelcast/issues/16513 for 4.1 milestone to track this.

Reply:
Maybe i'm missing something, but QueryCache indexes are local and should have no relation with partitioning subsystem.

Reply:
@ahmetmircik yes, but one can still call `addIndex` on one thread while running queries from another. Until `addIndex` wouldn't be finished with its job, queries would return inconsistent results.

Reply:
Thanks for clarification, now the opened issue makes sense.